### PR TITLE
Fix incorrect use of std::move() in g-api perf tests

### PR DESCRIPTION
**Pull request by aDanPin**
Replace 
    c.apply();
With
    cc = c.compile();
    cc();
In g-api performance tests.

Change gapi_core_perf_tests.cpp

### CODE REVIEW

**Comment by aDanPin** [comment]
@anton-potapov please review

**Comment by asmorkalov** [comment]
@anton-potapov Just a friendly reminder.

**Comment by aDanPin** [comment]
@anton-potapov , please review this pull request as fast as it possible.
I want to close this issue today

File: modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp [hunk 1]
```diff
@@ -36,16 +36,18 @@ PERF_TEST_P_(AddPerfTest, TestPerformance)
     cv::GComputation c(GIn(in1, in2), GOut(out));
 
     // Warm-up graph engine:
-    c.apply(gin(in_mat1, in_mat2), gout(out_mat_gapi), std::move(compile_args));
+    auto cc = c.compile(cv::descr_of(in_mat1), cv::descr_of(in_mat2), std::move(compile_args));
+    cc(in_mat1, in_mat2, out_mat_gapi);
```
Comment (anton-potapov) [inline modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp]: let's stick to common pattern : i.e. use `gin()` and `gout()` explicitly. 
i.e. 
```cpp
auto cc = c.compile(descr_of(gin(ginin_mat1, in_mat2)), std::move(compile_args));
cc(gin(in_mat1, in_mat2), gout(out_mat_gapi));
```

please note that no need to qualify `gin`, `gout` , `descr_of` with namespace (`cv::`) as they are in the same namespace as their actual argument (i.e. `cv::Mat`) and thus are found via [ADL](https://en.cppreference.com/w/cpp/language/adl)

Comment (anton-potapov) [reply modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp]: please apply this to all the changed calls

File: modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp [hunk 2]
```diff
@@ -36,16 +36,18 @@ PERF_TEST_P_(AddPerfTest, TestPerformance)
...
     TEST_CYCLE()
     {
-        c.apply(gin(in_mat1, in_mat2), gout(out_mat_gapi), std::move(compile_args));
+        cc(in_mat1, in_mat2, out_mat_gapi);
     }
 
     // Comparison ////////////////////////////////////////////////////////////
     // FIXIT unrealiable check: EXPECT_EQ(0, cv::countNonZero(out_mat_gapi != out_mat_ocv));
     EXPECT_EQ(out_mat_gapi.size(), sz);
+    //EXPECT_EQ(1, 0);
```
[the remark this answers is not in the document]
Comment (aDanPin) [reply modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp]: Removed

File: modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp [hunk 3]
```diff
@@ -175,11 +183,14 @@ PERF_TEST_P_(SubRCPerfTest, TestPerformance)
     cv::GComputation c(GIn(in1, sc1), GOut(out));
 
     // Warm-up graph engine:
-    c.apply(gin(in_mat1, sc), gout(out_mat_gapi), std::move(compile_args));
+    //auto cc = c.compile(gin(in_mat1, sc), gout(out_mat_gapi), std::move(compile_args));
```
[the remark this answers is not in the document]
Comment (aDanPin) [reply modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp]: Removed

File: modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp (hunk 1 above)
Comment (aDanPin) [reply modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp]: Fixed

File: modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp [hunk 4]
```diff
@@ -35,12 +35,17 @@ PERF_TEST_P_(AddPerfTest, TestPerformance)
...
     // Warm-up graph engine:
-    c.apply(gin(in_mat1, in_mat2), gout(out_mat_gapi), std::move(compile_args));
+    auto cc = c.compile(descr_of(gin(in_mat1, in_mat2)),
+                        std::move(compile_args));
+    cc(gin(in_mat1, in_mat2), gout(out_mat_gapi));
 
     TEST_CYCLE()
     {
-        c.apply(gin(in_mat1, in_mat2), gout(out_mat_gapi), std::move(compile_args));
+        cc(in_mat1, in_mat2, out_mat_gapi);
```
Comment (anton-potapov) [inline modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp]: let's use the same (i.e. main one) overload of `operator()` in performance test cycles as well. 
I.e. one that utilized easily with use of `gin()` and `gout()`. 

Please update rest of the path accordingly

File: modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp [hunk 5]
```diff
@@ -313,14 +332,16 @@ PERF_TEST_P_(DivPerfTest, TestPerformance)
     // G-API code ////////////////////////////////////////////////////////////
     cv::GMat in1, in2, out;
     out = cv::gapi::div(in1, in2, dtype);
-    cv::GComputation c(GIn(in1, in2), GOut(out));
+    cv::GComputation c(in1, in2, out);
```
Comment (anton-potapov) [inline modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp]: please revert this change, as unrelated to the issue

File: modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp (hunk 4 above)
Comment (aDanPin) [reply modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp]: Updated

File: modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp (hunk 5 above)
Comment (aDanPin) [reply modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp]: Fixed

**Comment by anton-potapov** [comment]
@alalek please merge it
